bin/stt_test/stt_providers/ISTTBase.py

```diff
@@ -0,0 +1,69 @@
+import json
+import logging
+from typing import List, Literal, Protocol, TypedDict
+import speech_recognition as sr  # type: ignore
+import time
+from stt_test.utils.SentenceChecker import SentenceChecker
+from stt_test.utils.test_data import get_audio_with_transcription
+import inspect
+
+
+class STTResult(TypedDict):
+    stt_trans: str
+    trans: str
+    similarity: float
+    execution_time: float
+
+
+class ISSTBase(Protocol):
+
+    MODEL_SIZES: List[str]
+    _sc: SentenceChecker
+    _recognizer: sr.Recognizer
+
+    def _init(self):
+        self._sc = SentenceChecker()
+        self._recognizer = sr.Recognizer()
+
+    def _before_all(self, model_size: str):
+        """Method for preparing models"""
+        ...
+
+    def _audio_to_text(self, audio: sr.AudioData, language: str) -> str:
+        """Methods that runs STT on a single audio"""
+        ...
+
+    def _save_results(self, data: List[STTResult], model: str):
+        stack = inspect.stack()
+        caller_class = stack[1][0].f_locals["self"].__class__.__name__
```

Comment:
simply `self.__class__.__name__` won't work in the same way?

Reply:
You're right, that's absolutely enough